src/home_robot/home_robot/core/interfaces.py

```diff
@@ -218,8 +218,18 @@ class Observations:
     semantic: Optional[
         np.array
     ] = None  # (camera_height, camera_width) in [0, num_sem_categories - 1]
+
+    # Instance IDs per observation frame
+    # Size: (camera_height, camera_width)
+    # Range: 0 to max int
+    instance: Optional[np.array] = None
```

Comment:
Should instance also have scores? That's needed for NMS. 
What about category ID?

Maybe "instance image" should be a dataclass bundling these together?

Reply:
this is just an instance mask prediction - it already has category segmentations

Reply:
I think this is actually not a bad idea, maybe we should add as an issue and do it when we need it?